### PR TITLE
Last discussions fixes

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR applies a lot of small fixes on the new discussion component:
- Fix the missing new discussion button
- Hide the form when the discussion has been created
- Focus (and expand) the newly created discussion after submit
- Display avatars in thread
- Fix missing first message
- Triggers auth earlier, when user the new comment or new discussion form instead of the submission (avoid loosing big messages on login)
- Fix some styling between the discussion header and the body
- Fix deep linking (on click and on load)
- Allows to close the comment and new discussion forms
- Removes some console warnings

(The first commit is not related and fixes the `clean` task)

### CODE REVIEW

File: js/components/discussions/threads-form.vue [hunk 1]
```diff
@@ -36,15 +35,13 @@ export default {
          * Prefill the form and focus the comment area
          */
         prefill(title, comment) {
+            comment = comment || '';
             this.title = title || '';
-            this.comment = comment || '';
+            this.comment = comment;
```
Comment (vinyll) [inline js/components/discussions/threads-form.vue]: I would move this line right below `comment = comment || '';`

File: js/components/discussions/threads.vue [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+    <a class="list-group-item add new-discussion" @click="displayForm" v-show="!formDisplayed">
         <div class="format-label pull-left">+</div>
-        <h4 class="list-group-item-heading">
-            {{ _('Start a new discussion') }}
-        </h4>
+        <h4 class="list-group-item-heading">{{ _('Start a new discussion') }}</h4>
     </a>
-    <div class="list-group-item list-group-form list-group-form-discussion animated"
-        v-show="formDisplayed" v-el:form>
+    <div v-el:form id="discussion-create" v-show="formDisplayed" v-if="currentUser"
```
Comment (vinyll) [inline js/components/discussions/threads.vue]: Should we then inform the anonymous user he can create a discussion if he were logged in?

File: js/components/discussions/threads.vue [hunk 3]
```diff
@@ -29,46 +27,121 @@
...
+         * Get the thread compoent for a given discussion id
+         */
+        threadFor(id) {
+            return this.$refs.threads.find($thread => $thread.discussion.id == id);
+        },
+
+        /**
+         * Handle deep linking
+         */
+        jumpToHash(hash) {
```
Comment (vinyll) [inline js/components/discussions/threads.vue]: Can't we use a simple `document.location.href`?

File: tasks.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
     'jpg', 'gif'
 )
 STATIC_ASSETS_DIRS = (
-    'dataset', 'organization', 'post', 'reuse', 'site', 'topic', 'user'
+    'dataset', 'organization', 'post', 'reuse', 'site', 'topic', 'user', 'chunks',
```
Comment (vinyll) [inline tasks.py]: Is this related to this PR?

File: js/components/discussions/thread-form.vue [hunk 5]
```diff
@@ -24,24 +22,32 @@ export default {
...
-          sending: false
+          sending: false,
+          comment: ''
       }
   },
   methods: {
+      /**
+       * Prefill the form and focus the comment area
+       */
+      prefill(comment) {
```
Comment (noirbizarre) [inline js/components/discussions/thread-form.vue]: These are not the same forms and there were already quite similar.
I can but it will be overly complex for nothing (a mixin or an intermediate component for both).

File: js/components/discussions/thread-form.vue [hunk 6]
```diff
@@ -24,24 +22,32 @@ export default {
...
-          }
           this.sending = true;
           this.$api
           .post(`discussions/${this.discussionId}/`, {comment: this.comment})
           .then(response => {
-              this.$dispatch('discussion-load', response);
+              this.$dispatch('discussion:updated', response);
               this.comment = '';
               this.sending = false;
+              document.location.href = `#discussion-${this.discussionId}-${response.discussion.length -1}`;
```
Comment (noirbizarre) [inline js/components/discussions/thread-form.vue]: 👍

File: js/components/discussions/threads-form.vue (hunk 1 above)
Comment (noirbizarre) [inline js/components/discussions/threads-form.vue]: it was written like that before but the comment is reused below twice and it needs to be a string everywhere so I factorised.

File: js/components/discussions/thread-form.vue (hunk 5 above)
Comment (vinyll) [inline js/components/discussions/thread-form.vue]: Ok to have some duplication over complexity.

File: js/components/discussions/threads.vue (hunk 3 above)
Comment (noirbizarre) [inline js/components/discussions/threads.vue]: Nop, the component is loaded after the hash trigger and so the ids are not present for resolution and scroll. And for comment or new comment links, the component needs to be expanded because you can't scroll to a non visible component.

File: tasks.py (hunk 4 above)
Comment (noirbizarre) [inline tasks.py]: Nop, this is why I precised it in the description. But I can split it into another PR.

File: js/components/discussions/threads-form.vue (hunk 1 above)
Comment (vinyll) [inline js/components/discussions/threads-form.vue]: I mean I'd rather read:
```
comment = comment || '';
this.comment = comment;
this.title = title || '';
```

File: js/components/discussions/threads.vue (hunk 2 above)
Comment (noirbizarre) [inline js/components/discussions/threads.vue]: Yes, but we should think/work on how we display and make it the same on every component because this is something new.
Until now, no component works this way, we never announce it needs login, we display on the login screen the reason that triggers it.

Right now, I just changed the moment where the auth is triggered.
Using YouCKAN, the login screen displays:
- you need to be authentified to post a comment
- you need to be authentified to create a discussion

File: tasks.py (hunk 4 above)
Comment (vinyll) [inline tasks.py]: Oh ok. It's a different commit anyways, so it's fine :)

File: js/components/discussions/threads.vue (hunk 2 above)
Comment (vinyll) [inline js/components/discussions/threads.vue]: Ok to consider this later in a more global overview

File: js/components/discussions/threads.vue (hunk 3 above)
Comment (vinyll) [inline js/components/discussions/threads.vue]: ok

File: js/components/discussions/thread-form.vue (hunk 5 above)
Comment (noirbizarre) [inline js/components/discussions/thread-form.vue]: 👍 👍

File: js/components/discussions/threads-form.vue (hunk 1 above)
Comment (noirbizarre) [inline js/components/discussions/threads-form.vue]: Ah OK ! I understand. I'll change it right now 👍

File: js/components/discussions/threads.vue (hunk 2 above)
Comment (noirbizarre) [inline js/components/discussions/threads.vue]: 👍 
We need to track the idea somewhere (issue ?)

File: tasks.py (hunk 4 above)
Comment (noirbizarre) [inline tasks.py]: ;)
